auto_nag/scripts/not_landed.py

```diff
@@ -300,21 +356,46 @@ def get_bugs(self, date="today", bug_ids=[]):
         bugs = self.filter_bugs(bugs)
         bugs_patch = self.get_patch_data(bugs)
         res = {}
+
+        reviewers_phid = set()
         nicknames = {}
+        for bugid, data in bugs_patch.items():
+            reviewers_phid |= data["reviewers_phid"]
+            assignee = bugs[bugid]["assigned_to"]
+            if not assignee:
+                assignee = max(data["author"], key=data["author"].get)
+                nicknames[bugid] = assignee
+
+        bz_reviewers = self.get_bz_userid(reviewers_phid)
+        nicknames = self.get_nicks(nicknames)
+
+        for data in bugs_patch.values():
+            rev = random.choice(list(data["reviewers_phid"]))
+            rev = bz_reviewers.get(rev)
+            if rev:
+                data["reviewer"] = rev
+
         for bugid, data in bugs_patch.items():
             res[bugid] = d = bugs[bugid]
             self.extra_ni[bugid] = data["count"]
             assignee = d["assigned_to"]
             nickname = d["nickname"]
+
             if not assignee:
-                assignee = max(data["author"], key=data["author"].get)
-                nicknames[bugid] = assignee
-            else:
-                self.add_auto_ni(bugid, {"mail": assignee, "nickname": nickname})
+                assignee, nickname = nicknames[bugid]
 
-        nicknames = self.get_nicks(nicknames)
-        for bugid, (name, nick) in nicknames.items():
-            self.add_auto_ni(bugid, {"mail": name, "nickname": nick})
+            if not assignee:
+                continue
+
+            flags = [utils.make_ni(assignee)]
+            reviewer = data.get("reviewer")
+            if reviewer:
+                flags.append(utils.make_ni(reviewer))
+
+            self.changes_per_bug[bugid] = {
```

Comment:
This is going to be overwritten by each iteration of the loop, isn't it?

Reply:
why ?

Reply:
No forget it, they are different bug IDs. I thought you were iterating the patches